### ISSUE TITLE
Automatically disable the system address book for large instances

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#49431]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**

As a Nextcloud admin I have to manually disable the system address book feature for my large installation because people are unable to use the Contacts app. 

**Describe the solution you'd like**

Automatically detect that the instance exceeded x users and turn the SAB off. Allow me to turn it on if I like.

**Describe alternatives you've considered**

Finish https://github.com/nextcloud/contacts/issues/3942.

**Additional context**

**Comment by ChristophWurst** [comment]
@SebastianKrupinski please backport this to stable31 (Hub 10)

**Comment by ChristophWurst** [comment]
This should also be noted in https://docs.nextcloud.com/server/latest/admin_manual/release_notes/upgrade_to_31.html so admins are not "surprised" by it.

**Comment by ChristophWurst** [comment]
Here are the refined requirements:
* The SAB is only turned off automatically at the upgrade from 31 to 32. All admins should receive a notification, making them aware of the automatic change.
* We add a setup check that warns admins when the threshold is exceeded. This will also be backported to help existing installations.

**Comment by SebastianKrupinski** [comment]
Should this be documented?

**Comment by ChristophWurst** [comment]
Yes, definitely, in the admin notes for 32: https://docs.nextcloud.com/server/latest/admin_manual/release_notes/upgrade_to_32.html

**Comment by SebastianKrupinski** [comment]
https://github.com/nextcloud/documentation/pull/13692
